### DISCUSSION TITLE
Captcha response = very long!

### DISCUSSION THREAD

**Discussion by jamms** [subdavis/kobo-book-downloader#79] [Q&A]
Hi, I've managed to get kobo-book-dowloader working on my Mac OS X (Monterey), but am falling at what feels like the final hurdle! Basically, when I am asked to paste the contents of the Captcha from Chrome into the terminal (I've used the native Mac OS X and also tried iTerm2), I can't paste the contents in as its too long. iTerm reports the length as 1,939 bytes in one line...

Looking at the example GIF that exists in the documentation, it only looks like 500 bytes or so. Has something changed with Kobo or am I doing something wrong here? Hopefully this screenshot will explain what I mean!

TIA

![Screenshot 2022-04-20 at 22 50 02](https://user-images.githubusercontent.com/1900406/164329711-c8d8bcb1-40e2-4c45-99bc-fbdea9e72d5f.png)

**Comment by subdavis** [comment]
You're definitely using an old version of kobodl.  You need to upgrade.  `kobodl --version` should be `0.9.2`.

**Reply by jamms** [reply]
Ah, thanks subdivision that's useful feedback! So I went to https://pypi.org/project/kobodl/ and copied the "pip install kobodl" from the top of the page. When installed, the version reports as "Successfully installed kobodl-0.8.0". However, on the webpage it is clearly 0.9.0. What am I doing wrong?

**Reply by subdavis** [reply]
You need to use the upgrade option.

pip install -U kobodl

**Reply by jamms** [reply]
Thanks, if only it was so easy! Command executes fine, outputs a lot of lines with dependencies 'Requirement already satisfied'. When I type in 'kobodl --version', I still get
kobodl, version 0.8.0

:(

**Reply by subdavis** [reply]
Sounds like you might just be having trouble with pip.  You can remove and re-install the package, clear your cache, use a virtual environment, or try one of the other installation options.

Incidentally, #80 is an issue now in both this project and its parent.  Kobo has broken our authentication again, so even if you got past this, you wouldn't be able to use either tool.

I'd honestly recommend the "Alternatives to kobodl" at the top of the README.  https://github.com/subdavis/kobo-book-downloader#alternatives-to-kobodl=

**Comment by jamms** [comment]
Hi subdavis, I see there's been some activity on the TnS-hun/kobo-book-downloader discussions in the last day to so and I can see you've updated to 0.9.2 on PyPI. I think the issue I'm having is down to pip not finding the latest version on PyPI and also an incompatibility with my version of Python. Would be interested in your views on the following outputs:

me@me0 % **pip install kobodl==0.9.2**
ERROR: Could not find a version that satisfies the requirement kobodl==0.9.0 (from versions: 0.2.2, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.3.6, 0.3.7, 0.4.0, 0.4.1, 0.5.0, 0.5.1, 0.6.0, 0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.8.0)
ERROR: No matching distribution found for kobodl==0.9.2

[same if I specify **kobodl==0.9.0** too]

me@me0 % **pip install https://files.pythonhosted.org/packages/36/73/a09b43f8e1f8ee6f7c9909451e0f273f5ca21877efc1b6fd73d0f50efcf1/kobodl-0.9.2.tar.gz**
Requirement already satisfied: click<8 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from kobodl==0.9.2) (7.1.2)
ERROR: Package 'kobodl' requires a different Python: 3.10.4 not in '<3.10,>=3.6'

**Reply by jamms** [reply]
Ok, so I fixed this by firing up an Ubuntu server on my MacBook using QEMU and everything works as expected with kobodl 0.9.2. Very happy as I now have my audiobooks. Many thanks!
